### PR TITLE
Refactor data format from plugin api

### PR DESCRIPTION
**Pull request by HaiDang666**
Metadata
- [x] Discogs
- [x] Audius
- [x] Musicbrainz
- [x] Bandcamp

Stream providers
- [x] Youtube
- [x] Audius
- [x] Bandcamp
- [x] Invidious
- [x] Soundcloud
- [x] Jamendo

### CODE REVIEW

File: packages/core/src/plugins/meta/discogs.ts [hunk 1]
```diff
@@ -146,10 +150,22 @@ class DiscogsMetaProvider extends MetaProvider {
...
-      .then(json => json.map(result => ({
-        ...result,
-        source: SearchResultsSource.Discogs
-      })));
+      .then(json => {
+        const artists = [];
+        const releases = [];
+
+        if (json.results) {
+          json.results.map(item => {
```
Comment (nukeop) [inline packages/core/src/plugins/meta/discogs.ts:158]: instead of pushing inside map, just assign the maps to consts (it's fine to map twice)

Comment (HaiDang666) [reply packages/core/src/plugins/meta/discogs.ts:158]: How to do it? I haven't done it before.

Comment (HaiDang666) [reply packages/core/src/plugins/meta/discogs.ts:158]: how ab this? cos  map return "undefined" if we don't return any
```js
  if (json.results) {
    const artists = json.results.flatMap(item => 
      (item.type === 'artist') ?
        [this.discogsArtistSearchResultToGeneric(item)] : []
    );

    const releases = json.results.flatMap(item =>
      (item.type === 'master' || item.type === 'release' ) ?
        [this.discogsReleaseSearchResultToGeneric(item)] : []
    );

    return Promise.resolve({ artists, releases, tracks: [] });
  }

  return Promise.resolve({ artists: [], releases: [], tracks: [] });
```

Comment (nukeop) [reply packages/core/src/plugins/meta/discogs.ts:158]: Yes, this is great.

In some situations you can use this: https://lodash.com/docs/4.17.15#compact

But your approach is better here

**Pull request merged**